File: Services/StartupService.cs
using System;
using System.Threading.Tasks;
using Microsoft.Extensions.DependencyInjection;
using DIExample;

[Service]
public class StartupService
{
    private readonly IServiceProvider _services;

    public StartupService(IServiceProvider services)
    {
        _services = Services;
    }

    public async Task InitialiseAsync() // If you have multiple services, it's good to have a Startup Service
    {
        Services.GetService<DatabaseService>().Initialise(); // so you can start all of them one by one organized.
        await Console.Out.WriteLineAsync("All services has been initialised");
        await Task.Delay(-1); // This is to avoid the program to finish.
    }
}

File: Services/StartupService.vb
Imports Microsoft.Extensions.DependencyInjection
<Service>
Public Class StartupService
    Private Services As IServiceProvider

    Sub New(Services As IServiceProvider)
        Me.Services = Services
    End Sub

    Public Async Function InitialiseAsync() As Task ' If you have multiple services, it's good to have a Startup Service
        Services.GetService(Of DatabaseService).Initialise() ' so you can start all of them one by one organized.
        Await Console.Out.WriteLineAsync("All services has been initialised")
        Await Task.Delay(-1) ' This is to avoid the program to finish.
    End Function
End Class
